File: packages/transform/src/features/http/CreateHttpValidateHeadersTransformer.ts
import { HttpValidateHeadersProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace CreateHttpValidateHeadersTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "http.createValidateHeaders",
      write: HttpValidateHeadersProgrammer.write,
    });
}

File: packages/transform/src/features/http/CreateHttpValidateHeadersTransformer.go
package http

import (
	corehttp "github.com/samchon/typia/packages/core/src/programmers/http"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var CreateHttpValidateHeadersTransformer = createHttpValidateHeadersTransformerNamespace{}

type createHttpValidateHeadersTransformerNamespace struct{}

func (createHttpValidateHeadersTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corehttp.HttpValidateHeadersProgrammer.Write(call.TypeText)
}
